vcenter_operator/configurator.py

```diff
@@ -316,21 +322,37 @@ def setup_os_session(self):
         )
         self.os_session = Session(auth=auth)
 
-    def _poll_nova(self):
-        if not self.os_session:
-            return
+    def _poll_nova_cells(self):
+        namespace_nova = self.global_options['namespace']
 
+        label_selector = 'system=openstack,component=nova,type=nova-cell'
         try:
-            endpoint_filter = {'service_type': 'compute', 'interface': 'public'}
-            resp = self.os_session.get('/os-cells', endpoint_filter=endpoint_filter)
-            for cell in resp.json().get('cellsv2', []):
-                self.global_options['cells'][cell['name']] = cell
-        except (HttpError, ConnectionError) as e:
-            LOG.error(f"Failed to get cells: {e}")
+            secrets = client.CoreV1Api().list_namespaced_secret(
+                namespace=namespace_nova,
+                label_selector=label_selector)
+        except client.ApiException as e:
+            LOG.error(f"Failed to retrieve secrets with labels {label_selector} from ns {namespace_nova}: {e}")
+            return False
+
+        if not secrets.items:
+            return
+
+        for secret in secrets.items:
+            if secret.type != 'Opaque':
+                LOG.error(f"Unknown secret type {secret.type} for {namespace_nova}/{secret.metadata.name}. "
+                          "Can only work with Opaque.")
+                return False
+
+            cell = {key: b64decode(value) for key, value in secret.data.items()}
+            self.global_options['cells'][cell['name']] = cell
```

Comment:
We should fail gracefully here as well, if the secret is somehow malformed and has no `'name'`.